File: AEDS2/Codes/C/TP01/Palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool Palindro(char s[], int i){
    if(i > (strlen(s)-1)){ // condicao de fim da recursao
        return true; // se sim, retorna true
    }
    if(s[i] != s[strlen(s)-1-i]){ // verifica se e diferente
        return false; // se sim, retorna false
    }

    return Palindro(s, i+1); // se nao cair na condicao de diferente, retorna a propria funcao para recursividade, ate correr todo a string
}

bool Palin(char s[]){   // cria a funcao para chamar a funcao Palindro
    return Palindro(s, 0);
}

int main(){
    char str[100];
    fgets(str, sizeof(str), stdin);
    str[strcspn(str, "\n")] = '\0'; // limpa o buffer

    while(strcmp(str, "FIM") != 0){ // cria o loop para correr as entradas
        if(strcmp(str, "FIM") != 0){ // condicao para evitar repeticoes a mais
            if(Palin(str)){ // condicao para chamar a funcao de palindromo e ver se e true
                printf("SIM\n"); // se true, imprime SIM
            }
            else{
                printf("NAO\n"); // se false, imprime NAO
            }
        }
        fgets(str, sizeof(str), stdin);
        str[strcspn(str, "\n")] = '\0'; // limpa o buffer
    }
}

File: AEDS2/Codes/Java/TP01/src/Palindromo.java
import java.util.Scanner;
//import java.util.*;

public class Palindromo {

    public static boolean Palindro(String str){ // cria a funcao de identificar um palindromo 
        int i = 0; 
        int j = str.length() - 1; 

        while(i < j){ // cria o loop para percorrer a string
            if(str.charAt(i) != str.charAt(j)){ // cria a condicao para verificar a string com sua inversa
                return false; // se houver diferenca ele retorna false
            }
            i++; 
            j--;
        }

        return true; // se nao cair nenhuma vez no if, quer dizer que e um palindromo, e retorna true
    }
    public static void main(String[] args) throws Exception {
         Scanner sc = new Scanner(System.in);
         String s = new String();
         s = sc.nextLine();

        while(!((s.length()) == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M')){ // cria o loop para correr todas as entradas
            if(!((s.length()) == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M')){ // condicao para evitar repeticao a mais
                if(Palindro(s)){ // condicao para chamar a funcao de palindromo e ver se e true
                    System.out.println("SIM"); // se sim, imprime SIM
                }
                else{
                    System.out.println("NAO"); // se nao, imprime NAO
                }
            }
            s = sc.nextLine();
        }

        sc.close();
    }
}
